controllers/manager.go

```diff
@@ -88,6 +95,40 @@ func NewManagerCommand() *cobra.Command {
 	return cmd
 }
 
+func preRunGarbageCollection(cl client.Client, logger *slog.Logger) error {
+	ctx := context.TODO()
+	req, err := labels.NewRequirement("openshiftVersion-major-minor", selection.In, []string{"4.16", "4.17", "4.18"})
+	if err != nil {
+		return fmt.Errorf("couldn't create ManagedCluster selector requirement. %w", err)
+	}
+
+	var managedClusters clusterv1.ManagedClusterList
+	err = cl.List(ctx, &managedClusters, &client.ListOptions{
+		LabelSelector: labels.NewSelector().Add(*req),
+	})
+	if err != nil {
+		return fmt.Errorf("couldn't list ManagedClusters. %w", err)
+	}
+	logger.Info(fmt.Sprintf("Found %v ManagedClusters matching given requirements", len(managedClusters.Items)), "Requirements", req)
+
+	for idx := range managedClusters.Items {
+		maintenanceAddon := addonapiv1alpha1.ManagedClusterAddOn{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "maintenance",
+				Namespace: managedClusters.Items[idx].Name,
+			},
+		}
+		err = cl.Delete(ctx, &maintenanceAddon, &client.DeleteOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete maintenance addon from cluster %q. %w", managedClusters.Items[idx].Name, err)
+		}
+		if !errors.IsNotFound(err) {
+			logger.Info("Deleted maintenance addon from ManagedCluster", "ManagedCluster", managedClusters.Items[idx].Name)
+		}
```

Comment:
```suggestion
		if err == nil || errors.IsNotFound(err) {
			logger.Info("Deleted maintenance addon from ManagedCluster", "ManagedCluster", managedClusters.Items[idx].Name)
		}
```
The condition above this is the same as this . 
This will never be hit. 
Most likely you meant to check if it is notFound or is nil

Reply:
Yeah good catch.